fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -712,54 +726,76 @@ private void handleQueryStmt(List<PlanFragment> fragments, List<ScanNode> scanNo
 
         coord.exec();
 
-        // send result
-        // 1. If this is a query with OUTFILE clause, eg: select * from tbl1 into outfile xxx,
-        //    We will not send real query result to client. Instead, we only send OK to client with
-        //    number of rows selected. For example:
-        //          mysql> select * from tbl1 into outfile xxx;
-        //          Query OK, 10 rows affected (0.01 sec)
-        //
-        // 2. If this is a query, send the result expr fields first, and send result data back to client.
         RowBatch batch;
-        MysqlChannel channel = context.getMysqlChannel();
-        boolean isOutfileQuery = false;
+        boolean isOutfileQuery;
         if (queryStmt instanceof QueryStmt) {
             isOutfileQuery = ((QueryStmt) queryStmt).hasOutFileClause();
-        } else if (queryStmt instanceof QueryStatement) {
+        } else {
```

Comment:
The change in StmtExecutor seems not elegant.

Reply:
Did you mean I should extract the process of outfileQuery judgment into individual functions?